### ISSUE TITLE
issue with Self keyword

### ISSUE THREAD

**Issue by awestlake87** [stepancheg/rust-protobuf#257]
when I generate this simple protobuf (excerpt taken from [s2client-proto](https://github.com/Blizzard/s2client-proto)), the Self identifier triggers a compile error in rust.

protobuf file:

```protobuf
enum Alliance {
  Self = 1;
  Ally = 2;
  Neutral = 3;
  Enemy = 4;
}
```

generated code (using programmatic protoc-rust in build.rs)

```rust
...
#[derive(Clone,PartialEq,Eq,Debug,Hash)]
pub enum Alliance {
    Self = 1,
    Ally = 2,
    Neutral = 3,
    Enemy = 4,
}
...
```

Is Self supposed to be suffixed with _pb since it's a keyword?

**Comment by stepancheg** [comment]
Enum values are not escaped now. Going to look how hard is to fix it.

**Comment by stepancheg** [comment]
I've created a patch (currently testing) which adds `value_` prefix to keywords, so `Self` becomes `value_Self`.

For a long time already field names which are rust keywords are prefixed with `field_`, so I followed this scheme.

Not sure about naming.

**Comment by stepancheg** [comment]
Published fix as 1.4.3.
